### ISSUE TITLE
error about "{chat[1:] if chat.startswith("@") else chat for chat in username}"

### ISSUE THREAD

**Issue by batianVolyc** [karfly/chatgpt_telegram_bot#129]
latest news,
I got error loop again

*****************************************
I modified the Dockerfile,
Made a change.
"FROM python:3.8-slim" > "FROM python:3.10.10-slim"
and it works ( not work )

I don't know why this error occurs, which link is wrong. Hope you guys can give me some guidance.
To try and fix this, I have tried it on two different vps (AMD64 and ARM64 Debian 11). All the same error.
I used default configuration and modified configuration (with my api key).

**********************Start**************************

chatgpt_telegram_bot  | Traceback (most recent call last):
chatgpt_telegram_bot  |   File "bot/bot.py", line 337, in <module>
chatgpt_telegram_bot  |     run_bot()
chatgpt_telegram_bot  |   File "bot/bot.py", line 314, in run_bot
chatgpt_telegram_bot  |     user_filter = filters.User(username=config.allowed_telegram_usernames)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 2323, in __init__
chatgpt_telegram_bot  |     super().__init__(chat_id=user_id, username=username, allow_empty=allow_empty)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 630, in __init__
chatgpt_telegram_bot  |     self._set_usernames(username)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 666, in _set_usernames
chatgpt_telegram_bot  |     self._usernames = self._parse_username(username)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 650, in _parse_username
chatgpt_telegram_bot  |     return {chat[1:] if chat.startswith("@") else chat for chat in username}
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 650, in <setcomp>
chatgpt_telegram_bot  |     return {chat[1:] if chat.startswith("@") else chat for chat in username}
chatgpt_telegram_bot  | AttributeError: 'int' object has no attribute 'startswith'
mongo                 | {"t":{"$date":"2023-03-14T05:05:14.143+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn3","msg":"Interrupted operation as its client disconnected","attr":{"opId":46}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:14.150+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn3","msg":"Connection ended","attr":{"remote":"172.18.0.3:48102","uuid":"0bd29f91-2ce9-4aa6-8861-6245c93cb7cd","connectionId":3,"connectionCount":2}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:14.181+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn4","msg":"Connection ended","attr":{"remote":"172.18.0.3:48108","uuid":"f6a68909-faed-4dd9-b763-d72563b44fee","connectionId":4,"connectionCount":1}}
chatgpt_telegram_bot exited with code 1
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.422+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.18.0.3:48114","uuid":"6cb30091-d4ca-4601-a105-979f4a647314","connectionId":5,"connectionCount":2}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.423+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn5","msg":"client metadata","attr":{"remote":"172.18.0.3:48114","client":"conn5","doc":{"driver":{"name":"PyMongo","version":"4.3.3"},"os":{"type":"Linux","name":"Linux","architecture":"x86_64","version":"5.10.0-20-amd64"},"platform":"CPython 3.8.16.final.0"}}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.429+00:00"},"s":"I",  "c":"NETWORK",  "id":22943,   "ctx":"listener","msg":"Connection accepted","attr":{"remote":"172.18.0.3:48130","uuid":"eab62785-6491-4905-80fd-ec0e17ac7c10","connectionId":6,"connectionCount":3}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.430+00:00"},"s":"I",  "c":"NETWORK",  "id":51800,   "ctx":"conn6","msg":"client metadata","attr":{"remote":"172.18.0.3:48130","client":"conn6","doc":{"driver":{"name":"PyMongo","version":"4.3.3"},"os":{"type":"Linux","name":"Linux","architecture":"x86_64","version":"5.10.0-20-amd64"},"platform":"CPython 3.8.16.final.0"}}}
chatgpt_telegram_bot  | Traceback (most recent call last):
chatgpt_telegram_bot  |   File "bot/bot.py", line 337, in <module>
chatgpt_telegram_bot  |     run_bot()
chatgpt_telegram_bot  |   File "bot/bot.py", line 314, in run_bot
chatgpt_telegram_bot  |     user_filter = filters.User(username=config.allowed_telegram_usernames)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 2323, in __init__
chatgpt_telegram_bot  |     super().__init__(chat_id=user_id, username=username, allow_empty=allow_empty)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 630, in __init__
chatgpt_telegram_bot  |     self._set_usernames(username)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 666, in _set_usernames
chatgpt_telegram_bot  |     self._usernames = self._parse_username(username)
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 650, in _parse_username
chatgpt_telegram_bot  |     return {chat[1:] if chat.startswith("@") else chat for chat in username}
chatgpt_telegram_bot  |   File "/usr/local/lib/python3.8/site-packages/telegram/ext/filters.py", line 650, in <setcomp>
chatgpt_telegram_bot  |     return {chat[1:] if chat.startswith("@") else chat for chat in username}
chatgpt_telegram_bot  | AttributeError: 'int' object has no attribute 'startswith'
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.926+00:00"},"s":"I",  "c":"-",        "id":20883,   "ctx":"conn5","msg":"Interrupted operation as its client disconnected","attr":{"opId":72}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.927+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn5","msg":"Connection ended","attr":{"remote":"172.18.0.3:48114","uuid":"6cb30091-d4ca-4601-a105-979f4a647314","connectionId":5,"connectionCount":2}}
mongo                 | {"t":{"$date":"2023-03-14T05:05:15.973+00:00"},"s":"I",  "c":"NETWORK",  "id":22944,   "ctx":"conn6","msg":"Connection ended","attr":{"remote":"172.18.0.3:48130","uuid":"eab62785-6491-4905-80fd-ec0e17ac7c10","connectionId":6,"connectionCount":1}}
chatgpt_telegram_bot exited with code 1
```

***********************End**************************

**Comment by itsupport-s2i** [comment]
it's run without any problem on mac os docker desktop, but not on ubuntu server.
same versions of docker and docker-compose

**Comment by karfly** [comment]
@batianVolyc 
Show me your `allowed_telegram_usernames` in `config/config.yml`

**Comment by karfly** [comment]
@itsupport-s2i are you sure that you get the same error?

@batianVolyc's error happens (I guess) because there is some int in `allowed_telegram_usernames ` (see `AttributeError: 'int' object has no attribute 'startswith'`)

**Comment by batianVolyc** [comment]
@karfly 
Thank you for reply.
Maybe I was wrong.
I had said to modify the version of python used by the Dockerfile. is my solution and it did work successfully. But when I try to add a mode in the chat_mode.yml file. Simple stop and start do not work. So I did the following two things and everything went into that error loop again.

```
docker-compose down
docker-compose --env-file config/config.env up --build
```

Regarding your question:
allowed_telegram_usernames: [] # if empty, the bot is available to anyone
I have tried the following three modes, none of which affect the occurrence of the error.
```
allowed_telegram_usernames: ["123456789"]
OR
allowed_telegram_usernames: [123456789]
OR
allowed_telegram_usernames: []
```
